seive.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+function seiveOriginal(n) {
+  let primes = new Array(n - 1).fill(true);
+  [primes[0], primes[1]] = [false, false];
+  for (let i = 2; i <= n; i++) {
+    for (let j = i * i; j <= n; j += i) {
+      primes[j] = false;
+    }
+  }
+
+  primes.forEach((el, i) => {
+    if (el) console.log(i);
+  });
+}
+
+function blocksSeive(n) {
+  let nPrime = new Array(n).fill(false);
+  let primes = [];
+  const sqrt = Math.sqrt(n);
+  for (let i = 2; i < sqrt; i++) {
+    if (!nPrime[i]) {
+      primes.push(i);
+      for (let j = i * i; j < n; j += i) {
+        nPrime[j] = true;
+      }
+    }
+  }
+
+  const s = 6;
+  let result = 0;
+  let nonPrime = new Array(s);
+
+  for (let k = 0; k <= Math.floor(n / s); k++) {
+    nonPrime.fill(false);
+    for (let i = 0; i < primes.length; i++) {
+      let start_id = (s * k) / primes[i];
+      if (start_id !== Math.floor(start_id))
+        start_id = Math.floor(start_id) + 1;
```

Comment:
Nice catch! I've forgotten this but you actually spotted this.
There is a math trick to remove the `if` check.
If you write `let start_id = Math.floor((s * k - 1) / primes[i]) + 1;`. This is equivalent to the 37..39 lines.

That works because `- 1` will affect the final result only when `s * k % primes[i] === 0`. And it will make the result of the division lesser by 1.

You should use the easiest alternative here. `if` clause here is perfectly fine. Just saying that there can be a math trick.

Resolve this comment if you think that's fine.